execution_engine/src/core/engine_state/genesis.rs

```diff
@@ -568,23 +746,26 @@ impl ExecConfig {
 
     /// Returns all bonded genesis validators.
     pub fn get_bonded_validators(&self) -> impl Iterator<Item = &GenesisAccount> {
-        self.accounts
-            .iter()
+        self.accounts()
             .filter(|&genesis_account| genesis_account.is_validator())
     }
 
     /// Returns all bonded genesis delegators.
-    pub fn get_bonded_delegators(
-        &self,
-    ) -> impl Iterator<Item = (&PublicKey, &PublicKey, &Motes, &Motes)> {
-        self.accounts
-            .iter()
+    pub fn get_bonded_delegators(&self) -> impl Iterator<Item = &DelegatorAccount> {
+        self.accounts()
             .filter_map(|genesis_account| genesis_account.as_delegator())
     }
 
     /// Returns all genesis accounts.
-    pub fn accounts(&self) -> &[GenesisAccount] {
-        self.accounts.as_slice()
+    pub(crate) fn accounts(&self) -> impl Iterator<Item = &GenesisAccount> {
+        self.accounts.iter()
+    }
+
+    /// Returns all genesis accounts.
```

Comment:
Copy and pasted comment:

```suggestion
    /// Returns all administrative accounts.
```